odd-platform-ui/src/components/shared/CopyButton/CopyButton.tsx

```diff
@@ -0,0 +1,67 @@
+import { CircularProgress } from '@mui/material';
+import React from 'react';
+import AppButton from '../AppButton/AppButton';
+import AppIconButton from '../AppIconButton/AppIconButton';
+import AlertIcon from '../Icons/AlertIcon';
+import CopyIcon from '../Icons/CopyIcon';
+import SuccessIcon from '../Icons/SuccessIcon';
+
+interface CopyButtonProps {
+  text?: string;
+  fallbackText?: string;
+  popupText?: string;
+  stringToCopy: string;
+  msDelay?: number;
+}
+const CopyButton: React.FC<CopyButtonProps> = ({
+  text = '',
+  fallbackText = 'Copying',
+  popupText = 'Copied!',
+  stringToCopy,
+  msDelay = 3000,
+}) => {
+  const [error, setError] = React.useState<string>('');
+  const [showCopy, setShowCopy] = React.useState<boolean>(false);
+  const [showCopying, setShowCopying] = React.useState<boolean>(false);
+  const copyToClipboard = async () => {
+    // copy(text)-> copying(fallbackText) -> copied(popupText) -> copy(text)
+    setShowCopying(true);
+    await navigator.clipboard
+      .writeText(stringToCopy)
+      .finally(async () => {
+        setShowCopying(false);
+        setShowCopy(true);
+        setTimeout(() => {
+          setShowCopy(false);
+          setError('');
+        }, msDelay);
+      })
+      .catch(() => setError('Copy error')); // no navigator.clipboard
```

Comment:
As we discussed you can simplify this function like this:
```const copyToClipboard = () => {
    if (navigator.clipboard) {
      setShowCopying(true);
      navigator.clipboard
        .writeText(stringToCopy)
        .then(() => {
          setShowCopying(false);
          setShowCopy(true);
        })
        .finally(() => {
          setTimeout(() => {
            setShowCopy(false);
            setError('');
          }, msDelay);
        })
        .catch(() => setError('Copy error'));
    }
    return '';
  };
```
  In this case you avoid unnecessary  async/await syntax.